framework/sysdeps/linux/frequency_manager.hpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef GENERIC_FREQUENCY_MANAGER_HPP
+#define GENERIC_FREQUENCY_MANAGER_HPP
+
+#ifndef __x86_64__
+#  include "../generic/frequency_manager.hpp"
+#else
+
+#include <limits>
+#include <fstream>
+#include <filesystem>
+#include <sandstone.h>
+
+#define DEFAULT_SYS_PATH        "/sys/devices/system/cpu/cpu"
+#define CPUINFO_MAX_FREQ_F      "cpuinfo_max_freq"
+#define CPUINFO_MIN_FREQ_F      "cpuinfo_min_freq"
+#define SCALING_MAX_FREQ_P      "/cpufreq/scaling_max_freq"
+#define KHZ                     100000
+
+class FrequencyManager
+{
+public:
+    FrequencyManager()
+    {
+        /* Discover supported and current frequencies */
+        std::filesystem::path cpuinfo_max_freq_p = "/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq";
+        max_freq_supported = get_frequency_from_file(cpuinfo_max_freq_p);
+
+        std::filesystem::path cpuinfo_min_freq_p = "/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_min_freq";
+        min_freq_supported = get_frequency_from_file(cpuinfo_min_freq_p);
+
+        std::filesystem::path scaling_max_freq_p = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq";
```

Comment:
For completeness of frequency control, FrequencyManager should also include support for uncore frequency scaling on supported processors (`/sys/devices/system/cpu/intel_uncore_frequency/`).

Reply:
For uncore frequency, I'd like to leave it for a future commit. But sure, we definitely should do it.